### PR TITLE
chore, refactor, fix, feat(egen): template structure+K80GPU+contract words+clean up

### PR DESCRIPTION
**Pull request by krishr2d2**
<br>
The notebook `multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb` uses K80 accelerator which is deprecated now., and follows the old template structure.

This PR includes the following updates:

1. Replace K80 with T4 GPUs.
2. Adds additional explanation at a few places.
3. Removes future tense.
4. Re-structures the notebook according to the new template.
5. Moves the setup related steps to the beginning.
6. Removes IS_TESTING from cleaning up section.
7. Adds steps to remove locally generated files in the cleaning up section.
8. Capitalizes the constants to run the training job.

<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

### CODE REVIEW

File: notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb [hunk 1]
```diff
@@ -77,18 +79,16 @@
...
+        "- Vertex AI Training\n",
         "\n",
         "The steps performed include:\n",
         "\n",
-        "- Building Custom Container using Artifact Registry and Docker\n",
-        "- Create a Vertex AI tensorboard instance to store your Vertex AI experiment\n",
-        "- Run a Vertex AI SDK CustomContainerTrainingJob"
+        "- Building a custom container using Artifact Registry and Docker.\n",
+        "- Creating a Vertex AI tensorboard instance to store your Vertex AI experiment.\n",
+        "- Run a Vertex AI training job using Python SDK."
```
Comment (kittyabs) [inline notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb:91]: Shouldn't "Python SDK" be "Vertex AI SDK for Python"?

File: notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb [hunk 2]
```diff
@@ -99,7 +99,7 @@
       "source": [
         "### Dataset\n",
         "\n",
-        "The dataset used for this tutorial is the <a href=\"http://yann.lecun.com/exdb/mnist/\">MNIST database</a>. The MNIST database of handwritten digits has a training set of 60,000 examples, and a test set of 10,000 examples. It is a subset of a larger set available from NIST. The digits have been size-normalized and centered in a fixed-size image.\n"
+        "The dataset used for this tutorial is the <a href=\"http://yann.lecun.com/exdb/mnist/\">MNIST database</a>. The MNIST database of handwritten digits has a training set of 60,000 examples, and a test set of 10,000 examples. It is a subset of a larger dataset constructed from  binary images of handwritten digits. The digits have been size-normalized and centered in a fixed-size image.\n"
```
Comment (kittyabs) [inline notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb:102]: Change "It is" to "This dataset"

File: notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb [hunk 3]
```diff
@@ -125,252 +125,193 @@
...
       "cell_type": "markdown",
       "metadata": {
-        "id": "create_bucket"
+        "id": "306bf9ab7f7f"
       },
       "source": [
-        "**Only if your bucket doesn't already exist**: Run the following cell to create your Cloud Storage bucket."
+        "### Initialize Vertex AI SDK for Python\n",
+        "\n",
+        "To get started using Vertex AI, you must have an existing Google Cloud project and [enable the Vertex AI API](https://console.cloud.google.com/flows/enableapi?apiid=aiplatform.googleapis.com)."
```
Comment (kittyabs) [inline notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb:301]: Change link text. Remove "enable the"
- "you must have an existing Google Cloud project and enable the [Vertex AI API]..."

File: notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb [hunk 4]
```diff
@@ -427,25 +368,30 @@
     {
       "cell_type": "markdown",
       "metadata": {
-        "id": "d48860504181"
+        "id": "2d242773d707"
       },
       "source": [
-        "### Import libraries and define constants"
+        "### Enable Artifact Registry API\n",
+        "You must enable the Artifact Registry API service for your project.\n",
```
Comment (kittyabs) [inline notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb:375]: Recommend adding a link to the documentation:
- You must enable the [Artifact Registry API](https://cloud.google.com/artifact-registry/docs/reference/rest) service for your project

File: notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb [hunk 5]
```diff
@@ -454,11 +400,9 @@
         "id": "12c03eca5908"
       },
       "source": [
-        "# Create Custom Training Python Package\n",
-        "\n",
-        "Before you can perform local training, you must create source code file, requirements file, docker file.\n",
+        "## Create training application\n",
         "\n",
-        "You will create a directory and write all of our files into that folder."
+        "Before you can perform local training, you create source code file, requirements file, and docker file for your training application."
```
Comment (kittyabs) [inline notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb:405]: Delete "you" (second one) and add "a" to the file noun phrases:
- Before you can perform local training, create a source code file, a requirements file, and a docker file for your training application.

File: notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb [hunk 6]
```diff
@@ -1076,7 +950,11 @@
         "id": "535932b8c42d"
       },
       "source": [
-        "## Executes in Workbench\n"
+        "## Executes in a notebook environment\n",
+        "\n",
+        "If you're running this tutorial in a notebook environment like Vertex AI workbench, proceed with the below steps. \n",
```
Comment (kittyabs) [inline notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb:955]: change "like" to "for example":

- If you're running this tutorial in a notebook environment, for example Vertex AI workbench, proceed with the below steps.

File: notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb [hunk 7]
```diff
@@ -1184,9 +1039,16 @@
         "id": "73c92c9298e9"
       },
       "source": [
-        "### Create a Vertex AI Tensorboard instance\n",
+        "## Create a Vertex AI Tensorboard instance\n",
+        "\n",
+        "Set a display name for your Tensorboard instance below and run the create command.\n",
+        "\n",
+        "**Note:**  To use an existing Vertex AI Tensorboard instance, you can use the following method.\n",
```
Comment (kittyabs) [inline notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb:1046]: remove "you can":
-  "To use an existing Vertex AI Tensorboard instance, use the following method."

File: notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb [hunk 8]
```diff
@@ -1354,17 +1222,26 @@
       },
       "outputs": [],
       "source": [
-        "# Set this to true only if you'd like to delete your bucket\n",
-        "delete_bucket = False\n",
-        "delete_tensorboard = False\n",
+        "# Delete the Training job\n",
```
Comment (kittyabs) [inline notebooks/official/training/multi_node_ddp_nccl_vertex_training_with_custom_container.ipynb:1225]: lower case for "training"
- "Delete the training job"

**Review by kittyabs — COMMENTED** [review]
several suggested edits

**Review by katiemn — APPROVED** [review]

**Pull request merged**